plugins/modules/dns_delegation.py

```diff
@@ -0,0 +1,332 @@
+                    returned: Always
+        disabled:
+            description:
+                - "Optional. I(true) to disable object. A disabled object is effectively non-existent when generating resource records."
+            type: bool
+            returned: Always
+        fqdn:
+            description:
+                - "Delegation FQDN. The FQDN supplied at creation will be converted to canonical form."
+                - "Read-only after creation."
+            type: str
+            returned: Always
+        id:
+            description:
+                - "The resource identifier."
+            type: str
+            returned: Always
+        parent:
+            description:
+                - "The resource identifier."
+            type: str
+            returned: Always
+        protocol_fqdn:
+            description:
+                - "Delegation FQDN in punycode."
+            type: str
+            returned: Always
+        tags:
+            description:
+                - "Tagging specifics."
+            type: dict
+            returned: Always
+        view:
+            description:
+                - "The resource identifier."
+            type: str
+            returned: Always
+"""  # noqa: E501
+
+from ansible_collections.infoblox.bloxone.plugins.module_utils.modules import BloxoneAnsibleModule
+
+try:
+    from bloxone_client import ApiException, NotFoundException
+    from dns_config import Delegation, DelegationApi
+except ImportError:
+    pass  # Handled by BloxoneAnsibleModule
+
+
+class DelegationModule(BloxoneAnsibleModule):
+    def __init__(self, *args, **kwargs):
+        super(DelegationModule, self).__init__(*args, **kwargs)
+
+        exclude = ["state", "csp_url", "api_key", "id"]
+        self._payload_params = {k: v for k, v in self.params.items() if v is not None and k not in exclude}
+        self._payload = Delegation.from_dict(self._payload_params)
+        self._existing = None
+
+    @property
+    def existing(self):
+        return self._existing
+
+    @existing.setter
+    def existing(self, value):
+        self._existing = value
+
+    @property
+    def payload_params(self):
+        return self._payload_params
+
+    @property
+    def payload(self):
+        return self._payload
+
+    def payload_changed(self):
+        if self.existing is None:
+            # if existing is None, then it is a create operation
+            return True
+
+        return self.is_changed(self.existing.model_dump(by_alias=True, exclude_none=True), self.payload_params)
+
+    def find(self):
+        if self.params["id"] is not None:
+            try:
+                resp = DelegationApi(self.client).read(self.params["id"], inherit="full")
+                return resp.result
+            except NotFoundException as e:
+                if self.params["state"] == "absent":
+                    return None
+                raise e
+        else:
+            filter = f"fqdn=='{self.params['fqdn']}'"
```

Comment:
Since view is not a required parameter for delegation, Not every user might create a custom view, some might use the Default view for the same and I think we can't add a filter if the view is not provided beforehand, hence it wasn't added to the  filter

Reply:
If we were to implement this , 
This could have two cases : Where user provides the view and when he doesnt 

Where user provides the view , this is sorted . 

But when view is no provided , we should assume that delegation is intended to be created in a zone in the default view . 
so we would have to make a view list call , provide default as a name filter and retrieve the view id.  
Optionally we can also check if the zone is present in the default view , coz if we dont check and zone isnt present , the backend would return "record not found".
Once this is verified , then the filter query would be executed with the view set

Reply:
Yes, I agree for idempotency, it should always include the view in the `find` filter. There are two approaches that you can take to it. 
- As you said, if `view` is not given by user, we can retrieve default view and use that ID. 
- Make view as a required parameter for all types of zones. Providing view_id of default view will have to be the user's responsibility in this case. I think this is much better, because the user can be explicit about which zone is being updated especially if there is one with the same fqdn in many views.

Reply:
Second method would definitely be better , but this would fail if the user fails to create a matching parent zone or has a typo in the delegation fqdn , coz if the matching Auth Zone isnt present in View. , then the backend would prompt the error "The delegation must have a parent zone" .

Reply:
We tried including view as a mandatory field and added it to the filters along with FQDN. below backend issue was noticed
https://infoblox.atlassian.net/browse/NORTHSTAR-12614

Reply:
Thanks for the quick backend fix, I've tested it and filter with view, and fqdn works